sky/packages/sky/lib/src/gestures/events.dart

```diff
@@ -0,0 +1,116 @@
+import 'dart:ui' as ui;
+
+/// Base class for input events.
+class InputEvent {
+
+  const InputEvent({ this.type, this.timeStamp: 0.0 });
+
+  factory InputEvent.fromUiEvent(ui.Event event) {
+    if (event is ui.PointerEvent)
+      return new PointerInputEvent.fromUiEvent(event);
+
+    // Default event
+    return new InputEvent(
+      type: event.type,
+      timeStamp: event.timeStamp
+    );
+  }
+
+  final String type;
+  final double timeStamp;
```

Comment:
Existing clients expect to be able to do math on this field; see e.g. https://github.com/flutter/engine/blob/master/sky/packages/sky/lib/src/gestures/drag.dart#L27. There might be others. In general I'm used to working with numeric timestamps on events rather than date/time objects, so I would lean towards leaving this as a double, but if you feel strongly I can look into converting the clients that expect to do math.